### PR TITLE
Feature/minor cleanups

### PR DESCRIPTION
**Pull request by r-englund**
See commit messages

### CODE REVIEW

**Review by martinfalk — APPROVED** [review]

File: modules/dataframe/src/datastructures/dataframe.cpp [hunk 1]
```diff
@@ -155,11 +155,13 @@ const std::vector<std::pair<std::string, const DataFormatBase *>> DataFrame::get
     return headers;
 }
 
-std::string DataFrame::getHeader(size_t idx) const { return columns_[idx]->getHeader(); }
+std::string DataFrame::getHeader(size_t idx) const { return columns_.at(idx)->getHeader(); }
```
Comment (petersteneteg) [inline modules/dataframe/src/datastructures/dataframe.cpp:158]: don't use exception in place of an assertion!

**Review by petersteneteg — CHANGES_REQUESTED** [review]

File: modules/dataframe/src/datastructures/dataframe.cpp (hunk 1 above)
Comment (r-englund) [reply modules/dataframe/src/datastructures/dataframe.cpp:158]: Assertions are only enabled in debug mode....

Comment (r-englund) [reply modules/dataframe/src/datastructures/dataframe.cpp:158]: or when we force enable them in cmake

Comment (petersteneteg) [reply modules/dataframe/src/datastructures/dataframe.cpp:158]: that is a different issue
You should not depend on exception for detecting programmer errors!

**Pull request closed without merging**